app/controllers/reactions_controller.rb

```diff
@@ -10,7 +10,7 @@ def add
     comment = nil
 
     if !comment_text.blank?
-      if @post.comments_disabled && !current_user.is_moderator && !current_user.is_admin
+      if @post.comments_disabled && !current_user.at_least_moderator?
```

Comment:
Untested suggestion:

```suggestion
      if @post.comments_disabled && current_user.standard?
```

Reply:
This might be one that's fine as there is no safe navigation - need to check whether that's not an oversight in the first place, though.

Reply:
As I mentioned in my other comments, I've decided to not touch the `current_user` ones for now even though I think it should be safe from the looks of it. We should probably revisit those at some point, though.